### DISCUSSION TITLE
Support for laravel livewire?

### DISCUSSION THREAD

**Discussion by dennykurniawan48** [spatie/laravel-pdf#154] [Q&A]
I tried to generate a pdf inside the laravel livewire render function and it showed the error:

`Call to undefined method Spatie\LaravelPdf\PdfBuilder::with()`

I have tried both the Pdf class and the pdf function. This is the code inside my livewire render component:

```
return pdf()->view('livewire.print.card', ['data' => $data])
                    ->name('datacard.pdf')
                    ->format(Format::A6)
                    ->download();
```

**Comment by H-Hp** [comment]
The error `Call to undefined method Spatie\LaravelPdf\PdfBuilder::with()` suggests that the `with()` method you're trying to use is not available in the version of the `spatie/laravel-pdf` package you're using or it has been removed or renamed.

Based on the code you provided, it seems you're trying to generate a PDF using the `spatie/laravel-pdf` package. Here's how you can generate a PDF inside the `render` function of your Livewire component:

```php
use Spatie\Pdf\Pdf;
use Spatie\Pdf\Dompdf;

public function render()
{
    $data = [...]; // Your data

    $pdf = Pdf::loadView('livewire.print.card', compact('data'))
        ->setPaper('a6')
        ->output();

    return $pdf->download('datacard.pdf');
}
```

Explanation:

1. First, make sure you have imported the necessary classes: `use Spatie\Pdf\Pdf;` and `use Spatie\Pdf\Dompdf;`.
2. Inside the `render` function, first prepare your data as usual.
3. Use `Pdf::loadView()` to load the view you want to render as a PDF, passing in the view name and any required data.
4. Use `setPaper('a6')` to set the paper size to A6. You can use other paper sizes as well, like 'a4', 'letter', etc.
5. Call `output()` to get the rendered PDF content.
6. Finally, return `$pdf->download('datacard.pdf')` to download the PDF with the specified filename.

Alternatively, if you want to return the PDF as a response instead of downloading it, you can use `$pdf->stream('datacard.pdf')` or `$pdf->inline('datacard.pdf')`.

Make sure you have installed the `spatie/laravel-pdf` package correctly and have set up the necessary configuration according to the package documentation.

If you're still facing issues, you can try updating the `spatie/laravel-pdf` package to the latest version, as the method names or usage might have changed in newer versions.
